### PR TITLE
Wallet block

### PR DESCRIPTION
**Pull request by lukestanley**
New Wallet block with previously demoed functionality.
Also recently since I merged servers for oauth and CORS proxying,  the structure of the requests changed, and real prod endpoints now exists, and some other small changes.
Also uses Block names in app UI.
Hopefully this is ready for prod time!

### CODE REVIEW

**Review by the-greenman — COMMENTED** [review]
My general comment on this may be related to the way we are using Angular. We are not really separating form and function in our work, and it is the source of many of the challenges we're facing in building working tools. 

I'd love to see us evolve this codebase to better align with the core Angular principle of using services for functionality and components for UI. 


I also am a fan of structured comments before each function. They help with shared coding and make nicer docs.

**Comment by lukestanley** [comment]
I've added comments for the main functions and put the Uphold service related parts into a service module, and it's much cleaner. @the-greenman

**Review by the-greenman — APPROVED** [review]
Sorry for all the hassle!

File: src/app/_shared/services/uphold_service.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+import { environment } from '../../../environments/environment';
+import * as uuid from 'uuid';
+const axios = require('axios').default;
+
+const UPHOLD_OAUTH_PROXY_URL = environment.uphold.proxy + 'api/login?code=';
+const UPHOLD_SITE_ROOT = environment.uphold.siteRoot;
+const UPHOLD_API_ROOT = environment.uphold.proxy;
+const MY_UPHOLD_CARDS = environment.uphold.proxy + 'v0/me/cards/';
+
```
[the remark this answers is not in the document]
Comment (lukestanley) [reply src/app/_shared/services/uphold_service.ts:12]: I can't do the {boolean} part due to the TSlint rule but the verb idea is good, I moved a bit. I like the style.

**Pull request merged**